src/platform/forms-system/src/js/review/ReviewCollapsibleChapter.jsx

```diff
@@ -83,208 +83,257 @@ class ReviewCollapsibleChapter extends React.Component {
+          <div key={arrayField.path} className="form-review-array">
+            <ArrayField
+              pageKey={page.pageKey}
+              pageTitle={page.title}
+              arrayData={_.get(arrayField.path, form.data)}
+              formData={form.data}
+              appStateData={page.appStateData}
+              formContext={formContext}
+              pageConfig={page}
+              onBlur={this.props.onBlur}
+              schema={arrayField.schema}
+              uiSchema={arrayField.uiSchema}
+              trackingPrefix={this.props.form.trackingPrefix}
+              setData={formData =>
+                this.props.setData(
+                  typeof page.updateFormData === 'function'
+                    ? page.updateFormData(form.data, formData)
+                    : formData,
+                )
+              }
+              path={arrayField.path}
+            />
+          </div>
+        ))}
+      </div>
+    );
+  };
+
+  getCustomPageContent = (page, props, editing) => {
+    if (editing) {
+      return (
+        <page.CustomPage
+          name={page.pageKey}
+          title={page.title}
+          trackingPrefix={props.form.trackingPrefix}
+          uploadFile={props.uploadFile}
+          onReviewPage
+          data={props.form.data}
+          updatePage={() => this.handleEdit(page.pageKey, false, page.index)}
+          pagePerItemIndex={page.index}
+        />
       );
     }
+    return (
+      <page.CustomPageReview
+        editPage={() => this.handleEdit(page.pageKey, !editing, page.index)}
+        name={page.pageKey}
+        title={page.title}
+        data={props.form.data}
+        pagePerItemIndex={page.index}
+      />
+    );
+  };
+
+  getChapterContent = props => {
+    const {
+      chapterFormConfig,
+      expandedPages,
+      form,
+      pageKeys,
+      viewedPages,
+    } = props;
+    const ChapterDescription = chapterFormConfig.reviewDescription;
+    return (
+      <div
+        className="usa-accordion-content schemaform-chapter-accordion-content"
+        aria-hidden="false"
+      >
+        {ChapterDescription && (
+          <ChapterDescription
+            viewedPages={viewedPages}
+            pageKeys={pageKeys}
+            formData={form.data}
+            t
+            push
+          />
+        )}
+        {expandedPages.map(page => {
+          const pageConfig = form.pages[page.pageKey];
+          const editing = pageConfig.showPagePerItem
+            ? pageConfig.editMode[page.index]
+            : pageConfig.editMode;
+
+          if (editing) {
+            return pageConfig.CustomPage
+              ? this.getCustomPageContent(page, props, editing)
+              : this.getSchemaformPageContent(page, props, editing);
+          } else {
+            return pageConfig.CustomPageReview
+              ? this.getCustomPageContent(page, props, editing)
+              : this.getSchemaformPageContent(page, props, editing);
+          }
```

Comment:
This is unfortunate. It looks like there are only two possible outcomes, but there are 4 paths (two for each)

What do you think about this (assuming I've done this right)?


```js

const showCustomPage = 
    (editing && pageConfig.CustomPage)
    || (!editing && pageConfig.CustomPageReview);

return showCustomPage 
    ? this.getCustomPageContent(page, props, editing)
    : this.getSchemaformPageContent(page, props, editing)
```

Reply:
Actually, reading yours, I think we can come up with a little better:
```js
const showCustomPage = editing
    ? !!pageConfig.CustomPage
    : !!pageConfig.CustomPageReview;

return showCustomPage 
    ? this.getCustomPageContent(page, props, editing)
    : this.getSchemaformPageContent(page, props, editing);
```